### PR TITLE
[BHT-126] Add pagination to subscriptions and users

### PR DESCRIPTION
**Pull request by StuditskyiMaksymITH**
[BHT-126] Add pagination to subscriptions and users

### CODE REVIEW

File: ability/views.py [hunk 1]
```diff
@@ -82,12 +74,28 @@ class AllWishViewSet(viewsets.ReadOnlyModelViewSet):
...
             Q(access_type='everyone') |
             Q(access_type='only_me', author=user) |
             Q(access_type='subscribers',
               author__in=Subscription.objects.filter(user=user).values_list('subscribed_to', flat=True))
-        ).exclude(author=user).distinct()
+        )
+        return queryset.exclude(author=self.request.user).distinct()
+
+    @action(detail=False, methods=['get'], url_path='(?P<user_id>\d+)')
+    def user_wish(self, request: Request, user_id=None, *args, **kwargs):
```
Comment (hordii-rushynets) [inline ability/views.py:87]: def user_wish(self, request: Request, user_id=None, *args, **kwargs):
        queryset = self.get_queryset().filter(author_id=user_id)
        serializer = self.get_serializer(queryset, many=True)
        return Response(serializer.data)

File: ability/views.py [hunk 2]
```diff
@@ -82,12 +74,28 @@ class AllWishViewSet(viewsets.ReadOnlyModelViewSet):
...
+        queryset = Wish.objects.filter(
             Q(access_type='everyone') |
             Q(access_type='only_me', author=user) |
             Q(access_type='subscribers',
               author__in=Subscription.objects.filter(user=user).values_list('subscribed_to', flat=True))
-        ).exclude(author=user).distinct()
+        )
+        return queryset.exclude(author=self.request.user).distinct()
+
+    @action(detail=False, methods=['get'], url_path='(?P<user_id>\d+)')
```
Comment (hordii-rushynets) [inline ability/views.py:86]: pagination_class

File: user/views.py [hunk 3]
```diff
@@ -157,3 +162,21 @@ def create(self, request: Request, *args, **kwargs) -> Response:
...
+class ListUserViewSet(viewsets.ReadOnlyModelViewSet):
+    """Return yser list view set"""
+    serializer_class = ShortBazhayUserSerializer
+    queryset = BazhayUser.objects.all()
+    permission_classes = [IsAuthenticated]
+    pagination_class = BazhayUserPagination
+    filter_backends = (DjangoFilterBackend,)
+    filterset_class = BazhayUserFilter
+
+    def get_queryset(self):
```
Comment (hordii-rushynets) [inline user/views.py:176]:
```
def get_serializer_class(self):
    serializers = {
         'list': ShortBazhayUserSerializer,
         'retrieve': UpdateUserSerializers
    }
    
    return serializers.get(self.action, ShortBazhayUserSerializer)
```

File: user/views.py [hunk 4]
```diff
@@ -157,3 +162,21 @@ def create(self, request: Request, *args, **kwargs) -> Response:
...
+        return self.queryset.exclude(id=self.request.user.id)
+
+    def retrieve(self, request, *args, **kwargs):
```
Comment (hordii-rushynets) [inline user/views.py:179]: to remove

File: user/serializers.py [hunk 5]
```diff
@@ -184,3 +191,15 @@ def create(self, validated_data: dict) -> BazhayUser:
...
+
+class ShortBazhayUserSerializer(serializers.ModelSerializer):
+    is_subscribed = serializers.SerializerMethodField()
+
+    class Meta:
+        model = BazhayUser
+        fields = ['id', 'photo', 'username', 'first_name', 'last_name', 'is_subscribed']
+        read_only_fields = ['id', 'photo', 'username', 'first_name', 'last_name', 'is_subscribed']
+
+    def get_is_subscribed(self, obj):
```
Comment (hordii-rushynets) [inline user/serializers.py:203]: add docstring

File: user/serializers.py [hunk 6]
```diff
@@ -184,3 +191,15 @@ def create(self, validated_data: dict) -> BazhayUser:
         except Exception:
             raise serializers.ValidationError({'error': 'Something went wrong'})
```
Comment (hordii-rushynets) [inline user/serializers.py:192]: detail='Something went wrong'

File: user/pagination.py [hunk 7]
```diff
@@ -0,0 +1,8 @@
+from rest_framework.pagination import PageNumberPagination
+
+
+class BazhayUserPagination(PageNumberPagination):
```
Comment (hordii-rushynets) [inline user/pagination.py:4]: Remove this pagination class. Use default PageNumberPagination

**Review by hordii-rushynets — COMMENTED** [review]
To fix

File: ability/views.py (hunk 1 above)
Comment (StuditskyiMaksymITH) [reply ability/views.py:87]: This code is no longer required

**Review by hordii-rushynets — APPROVED** [review]

**Pull request merged**